classes.py

```diff
@@ -0,0 +1,544 @@
+        else:
+            rating_gain = (1 + int(calc_rule == "DOUBLE_K")) * self.last_k * self.this_points_above_expected
+            rating_gain_rounded = round(rating_gain)  # Rounding to closest int
+            # self.new_rating = max(self.this_rating + rating_gain_rounded, 1)
+            self.new_rating = max(self.last_rating + rating_gain_rounded, 1)
+
+        print(str(self.id) + " | " + self.name + " | Old: " + str(self.last_rating) + " | New: " + str(self.new_rating))
+
+        # For debug
+        # print(str(self.id) + " | " + self.name + " | " + str(self.this_sum_oppon_ratings) + " | " + str(
+        #     self.this_expected_points))
+
+    def get_calculation_rule(self, is_fexerj_tournament):
+        if self.is_temp or self.is_unrated:
+            return "TEMPORARY"
+        elif self.check_rating_performance_rule() and is_fexerj_tournament:
+            return "RATING_PERFORMANCE"
+        elif self.check_double_k_rule():
+            return "DOUBLE_K"
+        return "NORMAL"
+
+    def check_rating_performance_rule(self):
+        if self.this_games < 5:
+            return False
+        elif self.this_games == 5:
+            return self.this_points_above_expected >= 1.84
+        elif self.this_games == 6:
+            return self.this_points_above_expected >= 2.02
+        elif self.this_games == 7:
+            return self.this_points_above_expected >= 2.16
+        raise NotImplementedError(
+            'Unknown condition for Rating performance rule with more than 7 games.')
+
+    def check_double_k_rule(self):
+        if self.this_games < 4:
+            return False
+        elif self.this_games == 4:
+            return self.this_points_above_expected >= 1.65
+        elif self.this_games == 5:
+            return self.this_points_above_expected >= 1.43
+        elif self.this_games == 6:
+            return self.this_points_above_expected >= 1.56
+        elif self.this_games == 7:
+            return self.this_points_above_expected >= 1.69
+        raise NotImplementedError(
+            'Unknown condition for double K rule with more than 7 games.')
+
+    def get_current_k(self):
+        # Assumes rating is not temporary
+        for (k, starting_num_games) in _K_STARTING_NUM_GAMES:
+            if self.last_total_games >= starting_num_games:
+                current_k = k
+        return current_k
+
+    def get_performance_rating(self, avg_oppon_rating, num_valid_games, total_num_points):
+        # In case of perfect results, consider score as if there was an extra game that ended in a
+        # draw.
+        score = total_num_points / num_valid_games
+        if score == 1.0:
+            score = (num_valid_games + 0.5) / (num_valid_games + 1.0)
+        elif score == 0.0:
+            score = 0.5 / (num_valid_games + 1.0)
+        return avg_oppon_rating + 400.0 * math.log10(score / (1.0 - score))
+
+
+class Tournament:
+
+    def __init__(self, rating_cycle, tournament):
```

Comment:
I really like how you pass the tournament argument and split below. Ideally we should do the same for rows above (see other comment above).